src/main/java/aquality/selenium/core/waitings/IConditionalWait.java

```diff
@@ -0,0 +1,81 @@
+package aquality.selenium.core.waitings;
+
+import org.openqa.selenium.support.ui.ExpectedCondition;
+
+import java.util.Collection;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BooleanSupplier;
+import java.util.function.Function;
+
+/**
+ * Utility used to wait for some condition.
+ */
+public interface IConditionalWait {
+
+    /**
+     * Wait for some condition within timeout. Method does not use WebDriverWait
+     * Default values for timeouts used from configuration settings file
+     *
+     * @param condition condition with boolean result (predicate)
+     * @param message   Part of error message in case of Timeout exception
+     * @return true if the condition has been met during the timeout
+     */
+    boolean waitFor(BooleanSupplier condition, String message);
+
+    /**
+     * Wait for some condition within timeout. Method does not use WebDriverWait
+     *
+     * @param condition                     condition with boolean result (predicate)
+     * @param timeoutInSeconds              Condition timeout
+     * @param pollingIntervalInMilliseconds Condition check interval
+     * @param message                       Part of error message in case of Timeout exception
+     * @return true if the condition has been met during the timeout
+     */
+    boolean waitFor(BooleanSupplier condition, long timeoutInSeconds, long pollingIntervalInMilliseconds, String message);
```

Comment:
can we make the seconds parameters to be nullable, as it is in aquality-selenium-core-dotnet? e.g. I would like to change the polling interval and want the timeout to remain the default; in the current implementation I would have to resolve the timeout configuration myself (meaning that I need to know that there is one). This is not very user-friendly I guess :)

Reply:
updated, could you check it please